### PR TITLE
refactor: move data availability validation out of state transition to allow optimistic sync

### PR DESCRIPTION
**Pull request by g11tech**
ref
 - https://github.com/ethereum/consensus-specs/pull/3185

### CODE REVIEW

**Review by dapplion — COMMENTED** [review]
While this was changed in the spec we don't really have to change anything right? I really like the need to have to explicitly state if the availability of the blob on import as it forces critical decisions publicly. Then if a block is not available it wont ever be imported to the forkchoice so it is not a concern to it

**Comment by g11tech** [comment]
> While this was changed in the spec we don't really have to change anything right? I really like the need to have to explicitly state if the availability of the blob on import as it forces critical decisions publicly. Then if a block is not available it wont ever be imported to the forkchoice so it is not a concern to it

I can accumulate and make the dataAvailable flags pass upstream to forkchoice's `onBlock` where we can check and throw on availability status to surface the considerations

i feel it doesnot belong to block state transition, also feel the same for execution status too that we pass there.

**Comment by wemeetagain** [comment]
Moving to draft and adding discussion tag

File: packages/beacon-node/src/api/impl/beacon/state/utils.ts [hunk 1]
```diff
@@ -254,7 +253,6 @@ async function getFinalizedState(
     state = stateTransition(state, block, {
       // Replaying finalized blocks, all data is considered valid
       executionPayloadStatus: ExecutionPayloadStatus.valid,
-      dataAvailableStatus: DataAvailableStatus.available,
```
Comment (dapplion) [inline packages/beacon-node/src/api/impl/beacon/state/utils.ts]: The main purpose of this notation was to force all consumers to explicitly declare assumptions about blobs. We will have to support different paths like regen block replay where we don't want to load blobs, but already know they are available. I'm aware it's technically pointless but being verbose about such critical decisions sounded good to me. I'm fine moving DA check outside of state transition, but this explicit statements should stay IMO

Comment (g11tech) [reply packages/beacon-node/src/api/impl/beacon/state/utils.ts]: ok sounds good, will make modifications

**Review by dapplion — DISMISSED** [review]

**Comment by g11tech** [comment]
rebased to resolve conflicts

**Review by dapplion — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

**Pull request merged**